File: 014.c
/**
    014.
    Ejercicio Nº 5 de la guía Nº 1
    
    @author Daniel Cantarín <canta@canta.com.ar>
    @author Bárbara Menares <fily@fily.com.ar>
    @date   20160514
*/

#include <stdio.h>
#include <math.h>

float p( float x );

int main() 
{
    
    float i;
    
    for (i = -1; i <= 1 ; i = i + 0.1)
    {
        printf( "p(%2.2f) = %2.2f.\n", i, p(i) );
    }
    
    
    return 0;
}


float p ( float x )
{
    float ret, tmp;
    
    ret = x * ( 2 / 3.1416);
    tmp = 1 - (powf(x,2.0) / 6) + ( powf(x,4.0) / 40 ) + ( powf(x, 6.0) / 336 ) + ( powf(x,8.0) / 3456) ;
    
    ret = ret * tmp;
    
    return ret;
}

File: 014.m
# Guia #4, Ej #3.
# @author   Daniel Cantarín <canta@canta.com.ar>
# @date     20161021

function salida = struct_file( f )
    
    x.dia   = 0;
    x.mes   = 0;
    x.anio  = 0;
    x.hora  = 0;
    
    x.tensiones  = [];
    x.corrientes = [];
    
    fid = fopen( f ,'r');
    ten = [];
    cor = [];
    i   = 0;
    
    fscanf( fid, '%s', 1 );
    fscanf( fid, '%s', 1 );
    tmp = fscanf( fid, '%s', 1 );
    [x.dia, x.mes, x.anio]  = strsplit( tmp, '-'){1,:};
    x.hora = fscanf( fid, '%s', 1 );
    fscanf( fid, '%s', 2 );
    
    while feof(fid) == 0
      i   = i + 1;
      tmp = fscanf( fid, '%f', 2);
      #tmp
      if size(tmp) > 0
        x.tensiones(i)   = tmp(1);
        x.corrientes(i)  = tmp(2);
      end
    end
    
    salida = x;
endfunction

